src/crimson/osd/pg.cc

```diff
@@ -705,7 +705,7 @@ template <class Ret, class SuccessFunc, class FailureFunc>
 PG::do_osd_ops_iertr::future<PG::pg_rep_op_fut_t<Ret>>
 PG::do_osd_ops_execute(
   OpsExecuter&& ox,
-  std::vector<OSDOp> ops,
+  std::vector<OSDOp>& ops,
```

Comment:
in `PG::do_osd_ops()`, this `m->ops` is passed to this method. and i am not sure if we are allowed move away from `m->ops`, as we might want to re-execute this message if we are interrupted by some event like PG interval change.

Reply:
?ops is not moved in do_osd_ops_execute

Reply:
`m->ops` is never moved, only "`ops`" of 
```
PG::do_osd_ops(ObjectContextRef obc,
  std::vector<OSDOp> ops,
  const OpInfo &op_info,
  const do_osd_ops_params_t& msg_params,
  do_osd_ops_success_func_t success_func,
  do_osd_ops_failure_func_t failure_func)
)
```
which is invoke in `InternalClientRequest::start()` is moved.

Reply:
per our discussion offline, we might need to

- update the commit message to explain the motivation of this change
- pass `ops` by const reference

Reply:
> * pass `ops` by const reference

@tchaikov It seems that, to accomplish this, we need to make multiple modifications to the current OpsExecuter, which may not be related to the purpose of this pr. I think I can submit a specific PR for this change. What do you think:-)

Reply:
@xxhdx1985126 sure. thanks in advance.

Reply:
@tchaikov You're welcome:-)